File: src/Java/Creational Patterns/Factory/CreditCards/src/Creators/MoneyBackCreditCardCreator.java
package Creators;

import Products.CreditCard;
import Products.MoneyBack;

import java.math.BigDecimal;

public class MoneyBackCreditCardCreator implements ICreditCardCreator {
    @Override
    public CreditCard createCreditCard(String model, BigDecimal limit, BigDecimal annualCharge) {
        return new MoneyBack(model, limit, annualCharge);
    }
}

File: src/JavaScript/CreationalPatterns/Factory/CreditCards/creators/moneyBackCreditCardCreator.js

const ICreditCardCreator = require('./creditCardCreator');
const MoneyBack = require('../products/moneyBack');

/**
 * @implements {ICreditCardCreator}
 */
class MoneyBackCreditCardCreator {
    /**
     * @inheritDoc
     */
    createCreditCard(model, limit, annualCharge) {
        return new MoneyBack(model, limit, annualCharge);
    }
}

module.exports = MoneyBackCreditCardCreator;